src/platform/vita/vita_input.c

```diff
@@ -0,0 +1,428 @@
+                            event->key.repeat = 0;
+                            break;
+                        case PAD_TRIANGLE:
+                            event->type = SDL_KEYDOWN;
+                            event->key.keysym.sym = SDLK_PAGEUP;
+                            event->key.keysym.mod = 0;
+                            event->key.repeat = 0;
+                            break;
+                        case PAD_CROSS:
+                        case PAD_R: // intentional fallthrough
+                            event->type = SDL_MOUSEBUTTONDOWN;
+                            event->button.button = SDL_BUTTON_LEFT;
+                            event->button.state = SDL_PRESSED;
+                            event->button.x = last_mouse_x;
+                            event->button.y = last_mouse_y;
+                            break;
+                        case PAD_CIRCLE:
+                        case PAD_L: // intentional fallthrough
+                            event->type = SDL_MOUSEBUTTONDOWN;
+                            event->button.button = SDL_BUTTON_RIGHT;
+                            event->button.state = SDL_PRESSED;
+                            event->button.x = last_mouse_x;
+                            event->button.y = last_mouse_y;
+                            break;
+                        case PAD_UP:
+                            event->type = SDL_KEYDOWN;
+                            event->key.keysym.sym = SDLK_UP;
+                            event->key.keysym.mod = 0;
+                            event->key.repeat = 0;
+                            up_pressed = 1;
+                            break;
+                        case PAD_DOWN:
+                            event->type = SDL_KEYDOWN;
+                            event->key.keysym.sym = SDLK_DOWN;
+                            event->key.keysym.mod = 0;
+                            event->key.repeat = 0;
+                            down_pressed = 1;
+                            break;
+                        case PAD_LEFT:
+                            event->type = SDL_KEYDOWN;
+                            event->key.keysym.sym = SDLK_LEFT;
+                            event->key.keysym.mod = 0;
+                            event->key.repeat = 0;
+                            left_pressed = 1;
+                            break;
+                        case PAD_RIGHT:
+                            event->type = SDL_KEYDOWN;
+                            event->key.keysym.sym = SDLK_RIGHT;
+                            event->key.keysym.mod = 0;
+                            event->key.repeat = 0;
+                            right_pressed = 1;
+                            break;
+                        case PAD_START:
+                            vkbd_requested = 1;
+                            break;
+                        default:
+                            break;
+                    }
+                }
+                break;
```

Comment:
Lots of redundant code in the switch statement. Could you refactor this?

You could for instance have an array containing pressed keys and indexed by `PAD_X` instead of `right_pressed`, `left_pressed`, etc. Then you can have a function taking `PAD_X` as input and doing the 5 statements. You can have another array to do the mapping between `PAD_X` and `SDLK_X`.

Reply:
Fixed in 35f42ab00e2b28a49e966a752eb3016a334d53bb